src/app/location-spoofing/LocationSpoofing.tsx

```diff
@@ -0,0 +1,149 @@
+import { useUnsealedResult } from '../../client/hooks/useUnsealedResult';
+import { getFlagEmoji, getIpLocation } from '../../shared/utils/locationUtils';
+import { getRegionalDiscount } from './data/getDiscountByCountry';
+import courseLogo from './fingerprintLogoLowOpacitySquareBordered.svg';
+import { env } from '../../env';
+import { TEST_IDS } from '../../client/testIDs';
+import { LOCATION_SPOOFING_COPY } from './copy';
+
+const COURSE_PRICE = 100;
+const TAXES = 15;
+const FALLBACK_DISCOUNT = 20;
+
+const LocationSpoofingUseCase: FunctionComponent = () => {
+  const { getData: getVisitorData, data: visitorData } = useVisitorData({
+    ignoreCache: true,
+  });
+  const { data: unsealedVisitorData } = useUnsealedResult(visitorData?.sealedResult);
+  let country;
+  let potentialDiscount;
+  if (unsealedVisitorData) {
+    country = getIpLocation(unsealedVisitorData)?.country;
+    potentialDiscount = country?.code ? getRegionalDiscount(country.code) : FALLBACK_DISCOUNT;
+  }
+
+  const {
+    mutate: activateRegionalPricing,
+    isLoading,
+    data: activateResponse,
+    error: activateError,
+  } = useMutation({
+    mutationKey: ['activate regional pricing'],
+    mutationFn: async () => {
+      const { requestId, sealedResult } = await getVisitorData({ ignoreCache: true });
+      const response = await fetch('/location-spoofing/api/activate-ppp', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({ requestId, sealedResult } satisfies ActivateRegionalPricingPayload),
+      });
+      return await response.json();
+    },
+    onSuccess: (data: ActivateRegionalPricingResponse) => {
+      if (data.severity === 'success') {
+        setDiscount(data.data.discount);
+      } else {
+        setDiscount(0);
+      }
+    },
+  });
+
+  const [courseCount, setCourseCount] = useState(1);
+  const [discount, setDiscount] = useState(0);
+
+  const cartItems = [
+    {
+      id: 0,
+      name: 'Fight Online Fraud Course',
+      subheadline: 'Fingerprint Inc.',
+      price: COURSE_PRICE,
+      image: courseLogo,
+      count: courseCount,
+      increaseCount: () => setCourseCount(courseCount + 1),
+      decreaseCount: () => setCourseCount(Math.max(1, courseCount - 1)),
+    },
+  ];
+
+  return (
+    <div className={classNames(styles.wrapper, formStyles.wrapper)}>
+      <Cart
+        items={cartItems}
+        discount={discount}
+        taxPerItem={TAXES}
+        discountLabel={LOCATION_SPOOFING_COPY.discountName}
+      ></Cart>
+      <div className={styles.innerWrapper}>
+        <form
+          onSubmit={(e) => {
+            e.preventDefault();
+            activateRegionalPricing();
+          }}
+          className={classNames(formStyles.useCaseForm, styles.regionalPricingForm)}
+        >
+          <p data-testid={TEST_IDS.locationSpoofing.callout}>
```

Comment:
If there is no country `name`, this reads as "We noticed you are from ! 👋 We are offering purchasing power parity pricing." Maybe could be rephrased when there is no country name available?

Reply:
Hmm good point, I assumed the country is always there if the identification data is there, but it doesn't have to. I modified the render condition to check for the country information specifically, thanks! 

https://github.com/fingerprintjs/fingerprintjs-pro-use-cases/pull/136/commits/4fa1d4ca2b3a1d6cfdae3b98d19e773a97c2468a